src/components/accessWrappers/ProjectRoute.tsx

```diff
@@ -0,0 +1,44 @@
+import { Navigate } from 'react-router-dom';
+
+import Loading from '../elements/Loading';
+import NotFound from '../pages/NotFound';
+
+import { ProjectPermissions } from '@/modules/permissions/types';
+import { useHasProjectPermissions } from '@/modules/permissions/useHasPermissionsHooks';
+import { useProjectDashboardContext } from '@/modules/projects/components/ProjectDashboard';
+import { ensureArray } from '@/utils/arrays';
+import { generateSafePath } from '@/utils/pathEncoding';
+
+/**
```

Comment:
This file is copied and only lightly modified from `EnrollmentProjectRoute.tsx`. It's also very similar to a few of the other files in that `accessWrappers` directory. Let me know if you would like me to dedicate some time to reducing repeated code here?

Reply:
looks good to me